src/discord_mcp/core/server/http_server.py

```diff
@@ -3,142 +3,67 @@
 
-if t.TYPE_CHECKING:
-    from discord_mcp.core.bot import Bot
+__all__: tuple[str, ...] = (
+    "DiscordMCPStarletteApp",
+    "HTTPDiscordMCPServer",
+)
 
 
 logger = logging.getLogger(__name__)
 
 
-class UserInfo(pydantic.BaseModel):
-    id: int
-    username: str
-    discriminator: str
-    avatar_url: t.Optional[str] = None
-
-
-def run_server(bot: Bot) -> None:
-    mcp = HTTPDiscordMCPServer(name="http-server", bot=bot)
-
-    # TODO: This will be replaced with a more dynamic tool loading system.
-    @mcp.tool()
-    async def get_latency(ctx: DiscordMCPContext) -> str:  # type: ignore[misc]
-        """Get the latency of the discord bot."""
-        return f"Latency: {ctx.request_context.lifespan_context.bot.latency * 1000:.2f} ms"
-
-    @mcp.tool()
-    async def get_bot_info(ctx: DiscordMCPContext) -> dict[str, t.Any]:  # type: ignore[misc]
-        """Get information about the discord bot."""
-        bot = ctx.request_context.lifespan_context.bot
-        if not bot.user:
-            logger.warning("Bot user is not available.")
-            raise ValueError("Bot user is not available.")
-        return bot.user._to_minimal_user_json()
-
-    @mcp.resource("resource://greeting")
-    def get_greeting() -> str:  # type: ignore[misc]
-        """Provides a simple greeting message."""
-        return "Hello from FastMCP Resources!"
-
-    # Resource returning JSON data (dict is auto-serialized)
-    @mcp.resource("data://config")
-    def get_config() -> dict[str, t.Any]:  # type: ignore[misc]
-        """Provides application configuration as JSON."""
-        return {
-            "theme": "dark",
-            "version": "1.2.0",
-            "features": ["tools", "resources"],
-        }
-
-    @mcp.resource("resource://system-status")
-    async def get_system_status(ctx: DiscordMCPContext) -> dict[str, t.Any]:  # type: ignore[misc]
-        """Provides system status information."""
-        return {
-            "status": "operational",
-            "request_id": ctx.request_id,
-            "uptime": f"{ctx.request_context.lifespan_context.bot.uptime.total_seconds()} seconds",
-        }
-
-    @mcp.resource("resource://user-info/{user_id}")
-    async def get_user_info(ctx: DiscordMCPContext, user_id: int) -> UserInfo:  # type: ignore[misc]
-        """Get information about a user by ID."""
-        bot = ctx.request_context.lifespan_context.bot
-        user = bot.get_user(user_id) or await bot.fetch_user(user_id)
-        if not user:
-            raise ValueError(f"User with ID {user_id} not found.")
-        return UserInfo(
-            id=user.id, username=user.name, discriminator=user.discriminator, avatar_url=user.display_avatar.url
+class DiscordMCPStarletteApp(Starlette):
+    def __init__(
+        self,
+        *args: t.Any,
+        bot: DiscordMCPBot,
+        mcp_server: HTTPDiscordMCPServer,
+        session_manager: StreamableHTTPSessionManager | None = None,
+        **kwargs: t.Any,
+    ) -> None:
```

Comment:
```suggestion
class DiscordMCPStarletteApp(Starlette):
    def __init__(
        self,
        bot: DiscordMCPBot,
        mcp_server: HTTPDiscordMCPServer,
        session_manager: StreamableHTTPSessionManager | None = None,
        *args: t.Any, 
        **kwargs: t.Any,
    ) -> None:
```

You were making all of the other arguments implicitly keyword only, it's not an error but not consistent and bad style